### ISSUE TITLE
Implement special case for m.room.canonical_alias

### ISSUE THREAD

**Issue by babolivier** [matrix-org/dendrite#618]
- [ ] Make sure the alias exists before allowing the event through
- [ ] Empty the event's content if the alias is removed

Requires a bit of plumbing but shouldn't be difficult.

**Comment by Kegsay** [comment]
Related sytests:
```
    × Canonical alias can be set
    × Canonical alias can include alt_aliases
    × Users with sufficient power-level can delete other's aliases
    × Can delete canonical alias
    × Only room members can list aliases of a room
```

**Comment by Lieunoir** [comment]
Hi, currently trying to get on this, however I'm unsure about what `alt_aliases` means. I considered this as a list of aliases that should be added if they don't already exist as aliases in the room, but looking at https://github.com/matrix-org/sytest/blob/develop/tests/30rooms/05aliases.pl#L147 I don't get why this test should fail : `$room_alias` can be set as a canonical alias, and `$bad_alias` is an alias that can be added.

**Comment by S7evinK** [comment]
Solved in #2236
